File: backend/app/models/recommendation.py
from sqlalchemy import Column, Integer, String, Text, DateTime
from sqlalchemy.sql import func

from app.core.database import Base


class Recommendation(Base):

    __tablename__ = "recommendations"

    id = Column(
        Integer,
        primary_key=True,
        index=True
    )

    sensor_id = Column(
        Integer,
        nullable=False,
        index=True
    )

    risk_level = Column(
        String(20),
        nullable=False,
        index=True
    )

    title = Column(
        String(255),
        nullable=False
    )

    description = Column(
        Text,
        nullable=False
    )

    action = Column(
        Text,
        nullable=True
    )

    created_at = Column(
        DateTime,
        server_default=func.now(),
        nullable=False
    )

File: frontend/src/types/recommendation.ts
import type { Sensitivity } from "./sensor";

export type RecommendationRisk =
  | "LOW"
  | "MEDIUM"
  | "HIGH"
  | "SEVERE"
  | string;

export interface Recommendation {
  id: number;
  sensor_id: number;

  risk_level: RecommendationRisk;

  title: string;
  description: string;

  action?: string | null;

  recommended_sensitivity?: Sensitivity | string | null;

  created_at: string;
}